### ISSUE TITLE
Add rationale for `rustc_type_ir` behind allowing rustc::potential-query-instability lint

### ISSUE THREAD

**Issue by ismailarilik** [rust-lang/rust#131318]

**Comment by compiler-errors** [comment]
It would be nice if you actually linked the specific callsites where this lint is triggering in the crate.

**Comment by ismailarilik** [comment]
I was just thinking that how that kind of `allow` statement looks like:

```rs
#[allow(rustc::potential_query_instability, "Order does not matter here.")]
```

I wished it to exist when I searched for these `allow` statements again and saw that a new statement is added without any reasoning. Of course, I've just ignored it because it is obvious that someone put it there for a _reason_.

But no one can ensure that this person didn't miss anything. So a _reasoning_ would always be helpful. So this proposition may be a way to enforce to put a reasoning before allowing a lint.

For now, I am just putting this here. If you don't think I am crazy, I will open an issue and propose this.

**Comment by ismailarilik** [comment]
> I was just thinking that how that kind of `allow` statement looks like:
> 
> #[allow(rustc::potential_query_instability, "Order does not matter here.")]

Oh, I wasn't the one who thinks this first: https://rust-lang.github.io/rfcs/2383-lint-reasons.html
